Frameworks/Foundation/NSOperation.mm

```diff
@@ -344,14 +347,14 @@ - (NSArray*)dependencies {
  @Status Interoperable
 */
 - (void)dealloc {
-    { // _dependenciesLock scope
-        std::lock_guard<std::recursive_mutex> lock(_dependenciesLock);
-        for (NSOperation* op in (NSSet*)_dependencies) {
-            [op removeObserver:self forKeyPath:@"isFinished" context:(void*)NSOperationContext];
-        }
+    for (NSOperation* op in (NSSet*)_dependencies) {
+        [op removeObserver:self forKeyPath:@"isFinished" context:(void*)NSOperationContext];
+    }
```

Comment:
nit: don't need to do this either if you haven't initialized.  also, a nil _dependencies *could* signify that you're not initialized.